src/classes/TDTM_ObjectDataGateway.cls

```diff
@@ -46,11 +46,11 @@ public with sharing class TDTM_ObjectDataGateway implements TDTM_iTableDataGatew
      * @return Boolean True is there are no TDTM records. False otherwise.
      */
     public Boolean isEmpty() {
```

Comment:
No, this needs to look directly at the saved records to decide if it should create our default ones.  If I just look at listTH.size() == 0, that will have loaded listTH with an empty list, and it won't get reset to the saved ones.  I could I guess check listTH.size() and then set it to null, but I'd rather just leave this method alone to its original code.